web/pandas/pdeps/0013-standardize-apply.md

```diff
@@ -0,0 +1,158 @@
+
+| func                    | Series     | DataFrame   |
+|:--------------------|:-----------|:------------|
+| lambda x: str(x)    | .map       | .map        |
+| lambda x: x + 1     | .transform | .transform  |
+| [lambda x: x.sum()] | .agg       | .agg        |
+
+So, for example, `ser.apply(lambda x: str(x))` can be replaced with `ser.map(lambda x: str(x))` while `df.apply([lambda x: x.sum()])` can be replaced with `df.agg([lambda x: x.sum()])`.
+
+Overall, because of their flexibility, `Series.apply` and `DataFrame.apply` are considered unnecessarily complex, and it would be clearer for users to use `.map`, `.agg` or `.transform`, as appropriate in the given situation.
+
+## Proposal
+
+With the above in mind, it is proposed that in the future `apply`, `transform` and `agg` will work as follows:
+
+1. the `agg` and `transform` methods of `Series`, `DataFrame` and `groupby` will always operate series-wise and never element-wise
+2. `Series.apply` and `DataFrame.apply` will be deprecated.
+3. `groupby.apply` will not be deprecated (because it behaves differently than `Series.apply` and `DataFrame.apply`)
+
+The above changes means that the future behavior, when users want to apply arbitrary callables in pandas, can be described as follows:
+
+1. When users want to operate on single elements in a `Series` or `DataFrame`, they should use `Series.map` and `DataFrame.map` respectively.
+2. When users want to aggregate a `Series`, columns/rows of a `DataFrame` or groups in `groupby` objects, they should use `Series.agg`, `DataFrame.agg` and `groupby.agg` respectively.
+3. When users want to transform a `Series`, columns/rows of a `DataFrame` or groups in `groupby` objects, they should use `Series.transform`, `DataFrame.transform` and `groupby.transform` respectively.
+4. Functions that are not applicable for  `map`, `agg` nor `transform` are considered relatively rare and in the future users should call these functions directly rather than use the `apply` method.
+
+The use of `Series.apply` and  `DataFrame.apply` will after the proposed change in almost all cases be replaced by `map`, `agg` or `transform`. In the very few cases where `Series.apply` and  `DataFrame.apply` cannot be substituted by `map`, `agg` or `transform`, it is proposed that it will be accepted that users will have to find alternative ways to apply the functions, i.e. typically apply the functions manually and possibly concatenating the results.
+
+It can be noted that the behavior of `groupby.agg`, `groupby.transform` and `groupby.apply` are not proposed changed in this PDEP, because `groupby.agg`, `groupby.transform` already behave as desired and `groupby.apply` behaves differently than `Series.apply` and `DataFrame.apply`. Likewise, the behavior when given ufuncs (e.g. `np.sqrt`) and string input (e.g. `"sqrt"`) will remain unchanged, because the behavior is already as intended in all cases.
```

Comment:
```suggestion
It can be noted that the behavior of `groupby.agg`, `groupby.transform` and `groupby.apply` are not proposed to be changed in this PDEP, because `groupby.agg`, `groupby.transform` already behave as desired and `groupby.apply` behaves differently than `Series.apply` and `DataFrame.apply`. Likewise, the behavior when given ufuncs (e.g. `np.sqrt`) and string input (e.g. `"sqrt"`) will remain unchanged, because the behavior is already as intended in all cases.
```